cloudstack/provider.go

```diff
@@ -94,6 +94,7 @@ func Provider() terraform.ResourceProvider {
 		ResourcesMap: map[string]*schema.Resource{
 			"cloudstack_affinity_group":       resourceCloudStackAffinityGroup(),
 			"cloudstack_autoscale_vm_profile": resourceCloudStackAutoScaleVMProfile(),
+			"cloudstack_cluster":              resourceCloudStackCluster(),
```

Comment:
great work @poddm. I'll try testing these changes.

Do we also need to add "cloudstack_zone" here ?

Reply:
@harikrishna-patnala  -  cloudstack_zone was partially added in a previous merge.  I extended the resource to support all of the fields and updated the ID from name to uuid.